# Problem №2

In [10]:
from sklearn.datasets import load_breast_cancer
from sklearn.datasets import load_digits 
breast_cancer = load_breast_cancer()
digits = load_digits()

In [20]:
print(breast_cancer.feature_names)
print(breast_cancer.data.view())
print(digits.data)

['mean radius' 'mean texture' 'mean perimeter' 'mean area'
 'mean smoothness' 'mean compactness' 'mean concavity'
 'mean concave points' 'mean symmetry' 'mean fractal dimension'
 'radius error' 'texture error' 'perimeter error' 'area error'
 'smoothness error' 'compactness error' 'concavity error'
 'concave points error' 'symmetry error' 'fractal dimension error'
 'worst radius' 'worst texture' 'worst perimeter' 'worst area'
 'worst smoothness' 'worst compactness' 'worst concavity'
 'worst concave points' 'worst symmetry' 'worst fractal dimension']
[[  1.79900000e+01   1.03800000e+01   1.22800000e+02 ...,   2.65400000e-01
    4.60100000e-01   1.18900000e-01]
 [  2.05700000e+01   1.77700000e+01   1.32900000e+02 ...,   1.86000000e-01
    2.75000000e-01   8.90200000e-02]
 [  1.96900000e+01   2.12500000e+01   1.30000000e+02 ...,   2.43000000e-01
    3.61300000e-01   8.75800000e-02]
 ..., 
 [  1.66000000e+01   2.80800000e+01   1.08300000e+02 ...,   1.41800000e-01
    2.21800000e-01   7.8200

In [21]:
from sklearn.naive_bayes import BernoulliNB, MultinomialNB, GaussianNB
from sklearn.model_selection import cross_val_score

In [34]:
def accuracy(data, typeNaivB):
    print cross_val_score(typeNaivB(), data.data, data.target).mean()

In [35]:
accuracy(breast_cancer, BernoulliNB)

0.627420402859


In [36]:
accuracy(breast_cancer, MultinomialNB)

0.894579040193


In [37]:
accuracy(breast_cancer, GaussianNB)

0.936749280609


In [38]:
accuracy(digits, BernoulliNB)

0.825823650778


In [39]:
accuracy(digits, MultinomialNB)

0.870877148974


In [40]:
accuracy(digits, GaussianNB)

0.818600380355


        1)Максимальное качество классификации на датасете breast_cancer: 0.936749280609
        2)Максимальное качество классификации на датасете digits: 0.870877148974\n
        3)С и D